### PR TITLE
7746 - Search bar redesign

### PR DESCRIPTION
**Pull request by latin-panda**
# Description

This PR apply the new design to Contacts and Reports' search bar. One thing to noticed is that the old design should be working fine as before, it can be enabled by adding `can_view_old_filter_and_search` [permission](https://docs.communityhealthtoolkit.org/apps/reference/app-settings/user-permissions/) to the config. 
- Creates a new Search Bar Component what is used in Contacts tab and Reports tab.
- Skips tour's step that explain about search, we did the same for the new filter in reports tab.
- Resets button is now only for filter, no for search, and it's located inside the filter side bar. 
- Adds a new button to clean search, a left chevron.
- Uses a new LESS variable `@media-small-mobile: 400px;` in all media query for small screens.
- The Sort stays pretty much the same as originally designed. The placement was fixed to match the new position of the tigger icon and the dropdown's text is darker to help readability. 
- Adds skip parameter to Global Actions `clearFilters`, this allows us to keep the value of a filter criteria, originally this was clearing search and filters, but now we just want to clear filters so we can skip "search" when clicking in "Reset". Any other situation (like ngDestroy of a tab page or the old filter) can clear both search and filters.

https://github.com/medic/cht-core/issues/7746
[CHT-Docs PR](https://github.com/medic/cht-docs/pull/776)

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by latin-panda** [comment]
Hi @m5r can you please review? Thanks for your time!

File: config/default/app_settings.json [hunk 1]
```diff
@@ -264,7 +264,8 @@
     ],
     "can_upgrade": [
       "program_officer"
-    ]
+    ],
+    "can_view_old_filter_and_search": []
```
Comment (m5r) [inline config/default/app_settings.json:268]: Nice! That makes it easy to play with this feature flag locally

**Review by m5r — APPROVED** [review]
LGTM, the new design looks good and it's nice to have a way to revert to the old one for partners who need more time to transition

File: webapp/src/css/variables.less [hunk 2]
```diff
@@ -86,6 +86,7 @@
 @content-margin: calc((100vw - @content-width) / 2);
 @admin-content-width: 1440px;
 @media-mobile: 767px;
+@media-small-mobile: 400px;
```
[the remark this answers is not in the document]
Comment (latin-panda) [reply webapp/src/css/variables.less:89]: Thank you!

File: config/default/app_settings.json (hunk 1 above)
Comment (latin-panda) [reply config/default/app_settings.json:268]: Thanks!

**Pull request merged**